internal/civisibility/utils/net/client.go

```diff
@@ -0,0 +1,228 @@
+		GetEarlyFlakeDetectionData() (*EfdResponseData, error)
+		GetCommits(localCommits []string) ([]string, error)
+		SendPackFiles(packFiles []string) (bytes int64, err error)
+	}
+
+	client struct {
+		id                 string
+		agentless          bool
+		baseUrl            string
+		environment        string
+		serviceName        string
+		workingDirectory   string
+		repositoryUrl      string
+		commitSha          string
+		branchName         string
+		testConfigurations testConfigurations
+		headers            map[string]string
+		handler            *RequestHandler
+	}
+
+	testConfigurations struct {
+		OsPlatform          string            `json:"os.platform,omitempty"`
+		OsVersion           string            `json:"os.version,omitempty"`
+		OsArchitecture      string            `json:"os.architecture,omitempty"`
+		RuntimeName         string            `json:"runtime.name,omitempty"`
+		RuntimeArchitecture string            `json:"runtime.architecture,omitempty"`
+		RuntimeVersion      string            `json:"runtime.version,omitempty"`
+		Custom              map[string]string `json:"custom,omitempty"`
+	}
+)
+
+func NewClient() Client {
+	ciTags := utils.GetCITags()
+
+	// get the environment
+	environment := os.Getenv("DD_ENV")
+	if environment == "" {
+		environment = "none"
+	}
+
+	// get the service name
+	serviceName := os.Getenv("DD_SERVICE")
+	if serviceName == "" {
+		if repoURL, ok := ciTags[constants.GitRepositoryURL]; ok {
+			// regex to sanitize the repository url to be used as a service name
+			repoRegex := regexp.MustCompile(`(?m)/([a-zA-Z0-9\\\-_.]*)$`)
+			matches := repoRegex.FindStringSubmatch(repoURL)
+			if len(matches) > 1 {
+				repoURL = strings.TrimSuffix(matches[1], ".git")
+			}
+			serviceName = repoURL
+		}
+	}
+
+	// get all custom configuration (test.configuration.*)
+	var customConfiguration map[string]string
+	if v := os.Getenv("DD_TAGS"); v != "" {
+		prefix := "test.configuration."
+		for k, v := range internal.ParseTagString(v) {
+			if strings.HasPrefix(k, prefix) {
+				if customConfiguration == nil {
+					customConfiguration = map[string]string{}
+				}
+
+				customConfiguration[strings.TrimPrefix(k, prefix)] = v
+			}
+		}
+	}
+
+	// create default http headers and get base url
+	defaultHeaders := map[string]string{}
+	var baseUrl string
+	var requestHandler *RequestHandler
+
+	agentlessEnabled := internal.BoolEnv(constants.CIVisibilityAgentlessEnabledEnvironmentVariable, false)
+	if agentlessEnabled {
+		// Agentless mode is enabled.
+		APIKeyValue := os.Getenv(constants.APIKeyEnvironmentVariable)
+		if APIKeyValue == "" {
+			log.Error("An API key is required for agentless mode. Use the DD_API_KEY env variable to set it")
+			return nil
+		}
+
+		defaultHeaders["dd-api-key"] = APIKeyValue
+
+		// Check for a custom agentless URL.
+		agentlessURL := ""
+		if v := os.Getenv(constants.CIVisibilityAgentlessURLEnvironmentVariable); v != "" {
+			agentlessURL = v
+		}
+
```

Comment:
Can we do:
```suggestion
		agentlessURL := os.Getenv(constants.CIVisibilityAgentlessURLEnvironmentVariable)
```
like for `APIKeyValue` above? The variable will end up with `""` anyway in case the variable is not defined.

Reply:
True, done in https://github.com/DataDog/dd-trace-go/pull/2869/commits/7c605497b6bc34274e08feabe9e21f98c2b4ef30